### PR TITLE
Critical improvements to service account commands

### PR DESCRIPTION
**Pull request by wtrocki**
## Issues covered

https://github.com/redhat-developer/app-services-cli/issues/1150
https://github.com/redhat-developer/app-services-cli/issues/1149
https://github.com/redhat-developer/app-services-cli/issues/1144

## Progress

- [x] Refactor Format of the Service acounts
- [x] Add auth url
- [x] Externalize auth URL with CLI Profiles (QA/Prod)

## Verification 

```
rhoas service-account create --name test --file-format=json --overwrite -v 
rhoas service-account create --name test --file-format=env --overwrite -v 
rhoas service-account create --name test --file-format=properties --overwrite -v 
```

The output should have valid `rhoas.service-account.oauthTokenUrl` URL:
```
## Generated by rhoas cli
rhoas.service-account.clientID=srvc-acct-f59de6ee-406e-4a0e-bbd6-25c6f8106db8
rhoas.service-account.clientSecret=a24a3d17-2ca9-4b67-b12c-192cfee1eca5
rhoas.service-account.oauthTokenUrl=https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token

```

### CODE REVIEW

File: pkg/cmd/serviceaccount/create/create.go [hunk 1]
```diff
@@ -144,6 +145,7 @@ func runCreate(opts *options) error {
 	creds := &credentials.Credentials{
 		ClientID:     serviceacct.GetClientId(),
 		ClientSecret: serviceacct.GetClientSecret(),
+		TokenURL:     "https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token",
```
Comment (craicoverflow) [inline pkg/cmd/serviceaccount/create/create.go:148]: Perhaps as a build variable?

File: pkg/serviceaccount/credentials/credentials.go [hunk 2]
```diff
@@ -68,7 +72,8 @@ func GetDefaultPath(outputFormat string) (filePath string) {
 // in the specified output format
 func Write(output string, filepath string, credentials *Credentials) error {
 	fileTemplate := getFileFormat(output)
-	fileBody := fmt.Sprintf(fileTemplate, credentials.ClientID, credentials.ClientSecret)
+	fileBody := fmt.Sprintf(fileTemplate, credentials.ClientID,
```
Comment (craicoverflow) [inline pkg/serviceaccount/credentials/credentials.go:75]: We should totally start using the template package at some point, maybe a Hacktoberfest issue?

https://pkg.go.dev/text/template

Comment (wtrocki) [reply pkg/serviceaccount/credentials/credentials.go:75]: IMHO this is very clean and simple for now. Templates will just externalize/complicate integration

File: pkg/cmd/serviceaccount/create/create.go (hunk 1 above)
Comment (wtrocki) [reply pkg/cmd/serviceaccount/create/create.go:148]: Yes.. I was pulled to other work. Doing that now

**Comment by wtrocki** [comment]
Ready for review

**Review by rkpattnaik780 — APPROVED** [review]

**Comment by wtrocki** [comment]
Thank you @rkpattnaik780

**Pull request merged**